Lib/fontTools/cffLib/__init__.py

```diff
@@ -164,44 +182,37 @@ def fromXML(self, name, attrs, content, otFont = None):
 
 		if name == "CFFFont":
 			if self.major == 1:
-				isCFF2 = False
+				self.cffCtx.isCFF2 = False
 				if not hasattr(self, "GlobalSubrs"):
-					self.GlobalSubrs = GlobalSubrsIndex()
+					self.GlobalSubrs = GlobalSubrsIndex(self.cffCtx)
 				if not hasattr(self, "fontNames"):
 					self.fontNames = []
-					self.topDictIndex = TopDictIndex()
+					self.topDictIndex = TopDictIndex(self.cffCtx)
 				fontName = attrs["name"]
 				self.fontNames.append(fontName)
-				topDict = TopDict(GlobalSubrs=self.GlobalSubrs)
+				topDict = TopDict(self.cffCtx, GlobalSubrs=self.GlobalSubrs)
 				topDict.charset = None  # gets filled in later
 			elif self.major == 2:
-				isCFF2 = True
+				self.cffCtx.isCFF2 = True
 				if not hasattr(self, "GlobalSubrs"):
-					self.GlobalSubrs = GlobalSubrsIndex2()
+					self.GlobalSubrs = GlobalSubrsIndex(self.cffCtx)
 				if not hasattr(self, "fontNames"):
 					self.fontNames = []
 				fontName = "CFF2Font"
 				self.fontNames.append(fontName)
 				cff2GetGlyphOrder = self.otFont.getGlyphOrder
-				topDict = TopDict2(GlobalSubrs=self.GlobalSubrs, cff2GetGlyphOrder=cff2GetGlyphOrder)
-				self.topDictIndex = TopDictData(None, cff2GetGlyphOrder, None)
+				topDict = TopDict(self.cffCtx, GlobalSubrs=self.GlobalSubrs, cff2GetGlyphOrder=cff2GetGlyphOrder)
+				self.topDictIndex = TopDictIndex(self.cffCtx, None, cff2GetGlyphOrder, None)
 			self.topDictIndex.append(topDict)
 			for element in content:
 				if isinstance(element, basestring):
 					continue
 				name, attrs, content = element
 				topDict.fromXML(name, attrs, content)
 		elif name == "GlobalSubrs":
-			if not isCFF2:
-				subrCharStringClass = psCharStrings.T2CharString
-				if not hasattr(self, "GlobalSubrs"):
-					self.GlobalSubrs = GlobalSubrsIndex()
-			elif isCFF2:
-				subrCharStringClass = psCharStrings.CFF2SubrCharString
-				if not hasattr(self, "GlobalSubrs"):
-					self.GlobalSubrs = GlobalSubrsIndex2()
-			else:
-				assert False, "Unsupported CFF format"
+			subrCharStringClass = psCharStrings.T2CharString
```

Comment:
We can rename it to CharString when we move it inside cffLib, and keep old module import it for compat.